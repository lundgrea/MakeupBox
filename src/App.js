import React, { Component } from 'react';
import './App.css'
import PropTypes from 'prop-types'
import { Route, NavLink } from 'react-router-dom'

import './App.css';

export class App extends Component {
  construtor() {
    super()
  }

  componentDidMount() {
  }

  render() {
    return (
      <main className="main">
      <h1>MAKEUP BOX</h1>
      </main>
      )
  }

}

export default App;
