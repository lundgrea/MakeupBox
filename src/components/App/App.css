* {
  box-sizing: border-box;
}

@font-face {
  font-family: "Source Sans Pro";
  src: url("https://fonts.googleapis.com/css?family=Source+Sans+Pro:300,400,700&display=swap");
}

a {
  text-decoration: none;
}

.main {
  margin-bottom: 10px;
}

