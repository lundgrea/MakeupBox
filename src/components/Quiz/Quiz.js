import React, { Component } from 'react';
import './Quiz.css';
import QuizQuestion from '../../components/QuizQuestion/QuizQuestion';

export class Quiz extends Component {
  constructor() {
    super()
    this.state = {
      questionOneAnswer: '',
      questionTwoAnswer: '',
      questionThreeAnswer: '',
      questionFourAnswer: '',
      quizQuestions: [
        {
          question: 'What is your primary interest?', 
          answers: ['lips', 'eyes','face/body', 'nails'],
          id: 1
        },
        {
          question: 'What style speaks most to you?', 
          answers: ['bronzed','bold', 'natural', 'smoky'],
          id: 2
        },
        { 
          question: 'What are your causes?',
          answers: ['cruelty free', 'organic', 'non-gmo', 'vegan'],
          id: 3
        }, 
        {
          question: 'What is your previous experience using makeup',
          answers: ['none', 'some', 'a bit', 'absolute master'],
          id: 4
        }
      ]    
    }
  }

  generateQuiz = () => {
    return this.state.quizQuestions.map(question => {
      return (
        <form>
          <QuizQuestion question={question.question} answers={question.answers} key={question.id} />
        </form>
      )
    })    
  }


  
  render() {
    return (
      <fieldset>
        {this.generateQuiz()}
      </fieldset>
    )
}


}

export default Quiz;