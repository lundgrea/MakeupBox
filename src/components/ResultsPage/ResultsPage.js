import React from 'react';
import './ResultsPage.css';

export const ResultsPage = () => {
  return (
    <h3>results</h3>
  )
}

export default ResultsPage