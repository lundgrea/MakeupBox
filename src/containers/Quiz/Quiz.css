fieldset {
  text-align: center;
  max-width: 700px;
  margin: auto;
  margin-top: 30px;
  border-radius: 2px;
  border: none;
  color: #242B28;
  background-color: #8EA99E;
}

label {
  margin: 0px 0px 0px 10px;
  font-size: 22px;
}

.button-quiz-submit {
  border: none;
  margin: 10px 10px 10px 10px;
  background-color: #242B28;
  color: #8EA99E;
  border: none;
  border-radius: 2px;
  width: 45%;
  height: 40px;
  font-size: 20px;
}

.button-quiz-submit:hover {
  color: white;
}

.quiz-headline {
  font-size: 30px;
}
