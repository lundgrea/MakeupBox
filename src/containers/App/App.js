import React, { Component } from 'react';
import './App.css';
import { fetchAllMakeup } from '../../apiCalls/apiCalls';
import { connect } from 'react-redux';
import { isLoading, handleErrors } from "../../actions";
import Home from '../../components/Home/Home';
import SelectorPage from '../../components/SelectorPage/SelectorPage';
import { Route, NavLink } from 'react-router-dom';

import PropTypes from 'prop-types'

import './App.css';

export class App extends Component {
  constructor () {
    super()
    this.state = {
      isLoading: true
    }
  }

  componentDidMount = async () => {
    // try {
    //   const allMakeup = await fetchAllMakeup();
    //   this.props.isLoading(false)
    //   this.setState({makeup: allMakeup, isLoading: false})

    // } catch ({message}) {
    //   this.props.handleErrors(message)
    //   this.props.isLoading(false)
    // }
  }

  render() {
    return (
      <main className="main">
        <Nav/>
      </main>
    )
  }
}









//   render() {
//     console.log(this.props)
//     return (
//       <main className="main">
//         <nav>
//           <h1>MAKEUP BOX</h1>
//           <NavLink to='/' className='nav'> Home </NavLink>
//           <NavLink to='/brands' className='nav'> Brands </NavLink>
//           <NavLink to='/makeup' className='nav'> Makeup </NavLink>
//           <NavLink to='/tags' className='nav'> Tags </NavLink>
//         </nav>
//         <section>
//           {this.props.error && <p className="error">{this.props.error}</p>}
//           {this.state.isLoading && <p className="loading">Page Is Loading</p>}
//         </section>
//         <section>
//           <Route exact path='/' component={Home} />
//           <Route path='/brands' component={SelectorPage} />
//           <Route path='/makeup' component={SelectorPage} />
//           <Route path='/tags' component={SelectorPage} />
//         </section>
//       </main>
//     )
//   }


export const mapStateToProps = store => ({
  error: store.hasErrored,
  isLoading: store.isLoading
})

export const mapDispatchToProps = dispatch => ({
  isLoading: bool => dispatch(isLoading(bool)),
  handleErrors: error => dispatch(handleErrors(error))
})

export default connect(mapStateToProps, mapDispatchToProps)(App);
