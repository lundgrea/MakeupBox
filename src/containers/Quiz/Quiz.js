import React, { Component } from 'react';
import './Quiz.css';
import QuizQuestion from '../../components/QuizQuestion/QuizQuestion';
import { setResponses } from "../../actions";
import { connect } from 'react-redux';
import { Link } from 'react-router-dom'

export class Quiz extends Component {
  constructor() {
    super()
    this.state = {
      userAnswers: [],     
      results: [], 
      quizQuestions: [
        {
          question: 'What is your primary interest?', 
          answers: ['lips', 'eyes','face/body', 'nails'],
          id: 1,
          name: 'interest'
        },
        {
          question: 'What style speaks most to you?', 
          answers: ['bronzed','bold', 'natural', 'smokey'],
          id: 2,
          name: 'style'
        },
        { 
          question: 'What are your causes?',
          answers: ['cruelty free', 'organic', 'non-gmo', 'vegan'],
          id: 3,
          name: 'causes'
        }, 
        {
          question: 'What is your previous experience using makeup',
          answers: ['none', 'some', 'a bit', 'absolute master'],
          id: 4,
          name: 'experience'
        }
      ]    
    }
  }

  generateQuiz = () => {
    return this.state.quizQuestions.map(question => {
      return (
        <QuizQuestion question={question.question} answers={question.answers} key={question.id} id={question.id} name={question.name} handleChange={this.handleChange}/>
      )
    })    
  }

  handleChange = (e) => {
    this.setState({userAnswers: [...this.state.userAnswers, e.target.value]})
    this.props.setResponses(e.target.value)
  }

  render() {
    return (
      <fieldset>
        {this.generateQuiz()}
        <Link to='/results'><button>submit answers</button></Link>
      </fieldset>
    )
  }
}

export const mapDispatchToProps = dispatch => ({
  setResponses: response => dispatch(setResponses(response))
});

export default connect(null, mapDispatchToProps)(Quiz);