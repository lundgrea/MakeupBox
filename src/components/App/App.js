import React, { Component } from 'react';
import './App.css';
import { Route } from 'react-router-dom';
import Nav from '../Nav/Nav';
import WelcomePage from '../WelcomePage/WelcomePage'
import Quiz from '../../containers/Quiz/Quiz'
import ResultsPage from '../../containers/ResultsPage/ResultsPage'

// import PropTypes from 'prop-types'


export class App extends Component {
  constructor () {
    super()
    this.state = {
      isLoading: true
    }
  }

  render() {
    return (
      <main className="main">
        <Nav />
        <Route exact path='/' component={WelcomePage} />
        <Route path='/quiz' component={Quiz} />
        <Route path='/results' component={ResultsPage} />
      </main>
    )
  }
}

export default App
