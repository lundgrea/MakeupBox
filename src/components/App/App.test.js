import React from 'react';
import { shallow, mount } from 'enzyme';
import { App } from './App';
import { MemoryRouter } from "react-router";
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import { Quiz } from '../../containers/Quiz/Quiz'
import { ResultsPage } from '../../containers/ResultsPage/ResultsPage'


describe('App', () => {
  it('should match the snapshot', () => {
    const wrapper = shallow(<App />);
    expect(wrapper).toMatchSnapshot();
  });
});

let mockStore, props, storeInstance;

beforeEach(()=> {
  mockStore = configureStore([])
  props = {
    hasErrored: '',
    isLoading: false,
    results: [],
    responses: []
  };
  storeInstance = mockStore(props)
})
describe('Router', () => {
  it('should show the Main page when nothing is selected', () => {
    const wrapper = mount(<Provider store={storeInstance}><MemoryRouter initialEntries={['/']}><App></App></MemoryRouter></Provider>)
    expect(wrapper.find(App)).toHaveLength(1);
  });
  
  it('should show the quiz page when Start Quiz is selected', () => {
    const wrapper = mount(<Provider store={storeInstance}><MemoryRouter initialEntries={['/quiz']}><App></App></MemoryRouter></Provider>)
    expect(wrapper.find(Quiz)).toHaveLength(1)
    });

  it('should show the results page once the quiz has been submitted', () => {
    const wrapper = mount(<Provider store={storeInstance}><MemoryRouter initialEntries={['/results']}><App></App></MemoryRouter></Provider>)
    expect(wrapper.find(ResultsPage)).toHaveLength(1)
  });

  it('should not show the results page once the Start Quiz button has been pressed', () => {
    const wrapper = mount(<Provider store={storeInstance}><MemoryRouter initialEntries={['/quiz']}><App></App></MemoryRouter></Provider>)
    expect(wrapper.find(ResultsPage)).toHaveLength(0)
  });

  it('should not show the quiz page once the submit quiz button has been pressed', () => {
    const wrapper = mount(<Provider store={storeInstance}><MemoryRouter initialEntries={['/results']}><App></App></MemoryRouter></Provider>)
    expect(wrapper.find(Quiz)).toHaveLength(0)
  });

  it('should not show the quiz or results page once the home button has been pressed', () => {
    const wrapper = mount(<Provider store={storeInstance}><MemoryRouter initialEntries={['/']}><App></App></MemoryRouter></Provider>)
    expect(wrapper.find(Quiz)).toHaveLength(0)
    expect(wrapper.find(ResultsPage)).toHaveLength(0)
  });

  it('should show the home page if some random url is entered', () => {
    const wrapper = mount(<Provider store={storeInstance}><MemoryRouter initialEntries={['/yolo']}><App></App></MemoryRouter></Provider>)
    expect(wrapper.find(App)).toHaveLength(1)
  });

  it('should not show the quiz or results page if some random url is entered', () => {
    const wrapper = mount(<Provider store={storeInstance}><MemoryRouter initialEntries={['/yolo']}><App></App></MemoryRouter></Provider>)
    expect(wrapper.find(Quiz)).toHaveLength(0)
    expect(wrapper.find(ResultsPage)).toHaveLength(0)  
  });
});

