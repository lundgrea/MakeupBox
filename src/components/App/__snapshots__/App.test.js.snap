// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot 1`] = `
<main
  className="main"
>
  <Nav />
  <Route
    component={[Function]}
    exact={true}
    path="/"
  />
  <Route
    component={
      Object {
        "$$typeof": Symbol(react.memo),
        "WrappedComponent": [Function],
        "compare": null,
        "displayName": "Connect(Quiz)",
        "type": [Function],
      }
    }
    path="/quiz"
  />
  <Route
    component={
      Object {
        "$$typeof": Symbol(react.memo),
        "WrappedComponent": [Function],
        "compare": null,
        "displayName": "Connect(ResultsPage)",
        "type": [Function],
      }
    }
    path="/results"
  />
</main>
`;
